### PR TITLE
Partially fixed #16: Introduced fragment components for profile editing in settings

### LINKED ISSUE #16: Implement multi-user profile management system [Blocked: #14, #167]
A profile system needs to be implemented satisfying the requirements described in the Oppia PRD. This system should support storing a context of the currently logged-in profile, and support logging out that profile or logging in another profile. This system is responsible for the pin-based authentication described in the PRD.

Other systems might need to be updated to rely on this context rather than requiring a profile ID to be passed in to simplify them.

Note that for the prototype, it's essential that at least some portion of this system be implemented even if it just defaults to a single user without support for profile switching. Supporting profiles plays a central role in the design for the entire app architecture, so adding it late will be complex.

### PR DESCRIPTION
**Pull request by jamesxu0**
## Explanation

Partially fixing #16: Introduced fragment components for profile editing in settings. Introduces ProfileListFragment, ProfileEditFragment, AdminAuthFragment, and AddProfileFragment.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is **assigned** to an appropriate reviewer.

### CODE REVIEW

**Comment by BenHenning** [comment]
Hi @jamesxu0. Deferring this review for now until my comment in your document is resolved about whether these should actually be fragments vs. activities. Please assign back to me once that's finalized & this PR is up-to-date (if changes are needed).

**Comment by jamesxu0** [comment]
@BenHenning PTAL. Changed ProfileRenameFragment and ProfileResetPinFragment to ProfileRenameActivity and ProfileResetPinActivity.

File: app/src/main/java/org/oppia/app/settings/profile/ProfileDeletionDialogFragment.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.oppia.app.settings.profile
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/settings/profile/ProfileDeletionDialogFragment.kt]: If we aren't using this in this PR, can we remove it and add it later?

File: app/src/main/java/org/oppia/app/settings/profile/ProfileListFragment.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import org.oppia.app.fragment.InjectableFragment
+import javax.inject.Inject
+
+/** Fragment that allows user to select a profile to edit from settings. */
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/settings/profile/ProfileListFragment.kt]: Nit: ``that allows users to select``

File: app/src/main/java/org/oppia/app/settings/profile/ProfileListFragmentPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import javax.inject.Inject
+
+/** The presenter for [ProfileListFragment]. */
+@FragmentScope
+class ProfileListFragmentPresenter @Inject constructor(
+  private val fragment: Fragment,
+  private val viewModelProvider: ViewModelProvider<ProfileListViewModel>
+) {
+  fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
+    val binding = ProfileListFragmentBinding.inflate(inflater, container, /** attachToRoot= */ false)
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/settings/profile/ProfileListFragmentPresenter.kt]: Nit: Comment should be ``/* attachToRoot= */``. '/**' is for KDocs.

Ditto elsewhere.

File: app/src/main/res/layout/profile_deletion_dialog.xml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (BenHenning) [inline app/src/main/res/layout/profile_deletion_dialog.xml]: Ditto for this file--let's remove if we aren't using it anywhere in this PR.

**Review by BenHenning — APPROVED** [review]
Thanks.

File: app/src/main/java/org/oppia/app/settings/profile/ProfileDeletionDialogFragment.kt (hunk 1 above)
Comment (jamesxu0) [reply app/src/main/java/org/oppia/app/settings/profile/ProfileDeletionDialogFragment.kt]: Removed.

File: app/src/main/java/org/oppia/app/settings/profile/ProfileListFragment.kt (hunk 2 above)
Comment (jamesxu0) [reply app/src/main/java/org/oppia/app/settings/profile/ProfileListFragment.kt]: Updated.

File: app/src/main/java/org/oppia/app/settings/profile/ProfileListFragmentPresenter.kt (hunk 3 above)
Comment (jamesxu0) [reply app/src/main/java/org/oppia/app/settings/profile/ProfileListFragmentPresenter.kt]: Updated.

File: app/src/main/res/layout/profile_deletion_dialog.xml (hunk 4 above)
Comment (jamesxu0) [reply app/src/main/res/layout/profile_deletion_dialog.xml]: Removed.

**Pull request merged**